LEGO1/mxeventpresenter.cpp

```diff
@@ -44,10 +47,71 @@ void MxEventPresenter::Destroy()
 
 	m_criticalSection.Enter();
 
-	if (m_unk50)
-		delete m_unk50;
+	if (m_data)
+		delete[] m_data;
 
 	Init();
 
 	m_criticalSection.Leave();
 }
+
+// OFFSET: LEGO1 0x100c2e30
+void MxEventPresenter::CopyData(MxStreamChunk* p_chunk)
+{
+	m_data = new MxU8[p_chunk->GetLength()];
+	memcpy(m_data, p_chunk->GetData(), p_chunk->GetLength());
```

Comment:
m_data -> m_chunk?

Reply:
`m_data` is probably a custom structure, but there isn't sufficient information right now to figure out what it is exactly. Until then I think that `m_data` is an appropriate name since for now it's based on a verbatim copy of `p_chunk->m_data`.

Reply:
Might be some sort of dictionary or Hashmap

Reply:
This data comes from the `.si` files, and since @itsmattkc has a library documenting the types the structure can probably be extracted from it: https://github.com/itsmattkc/SIEdit/tree/master/lib

I haven't looked into it yet however, but the information is almost certainly there.